### PR TITLE
Filter courses based on organization of current sites before serving

### PR DESCRIPTION
**Pull request by iam-mhaseeb**
**Description**: Filtered courses based on the organization of current sites before serving
**Jira**: https://edlyio.atlassian.net/browse/EDLY-1811

**Testing**:
Login with course_creator role on multiple sites to verify they have courses of their sites only.

### CODE REVIEW

File: openedx/features/edly/utils.py [hunk 1]
```diff
@@ -290,3 +290,29 @@ def user_can_login_on_requested_edly_organization(request, user):
         return True
 
     return False
+
+
+def get_all_courses_based_on_org(request, all_courses):
```
Comment (zubair-arbi) [inline openedx/features/edly/utils.py:295]: `filter_courses_based_on_org`

File: openedx/features/edly/utils.py [hunk 2]
```diff
@@ -290,3 +290,29 @@ def user_can_login_on_requested_edly_organization(request, user):
...
+    Arguments:
+        request: HTTP request object,
+        all_courses (list): List of CourseAccessRole objects.
+
+    Returns:
+        list: Returns List of courses filtered based on current site organization.
+    """
+
+    current_site = request.site
+    edly_sub_org = EdlySubOrganization.objects.get(
```
Comment (rehan99000) [inline openedx/features/edly/utils.py:311]: why make a query if we can get edly sub org by `current_site.edly_sub_org` ?

**Review by rehan99000 — COMMENTED** [review]
One query otherwise looks good

File: openedx/features/edly/utils.py [hunk 3]
```diff
@@ -290,3 +290,29 @@ def user_can_login_on_requested_edly_organization(request, user):
...
+
+    Arguments:
+        request: HTTP request object,
+        all_courses (list): List of CourseAccessRole objects.
+
+    Returns:
+        list: Returns List of courses filtered based on current site organization.
+    """
+
+    current_site = request.site
```
Comment (zubair-arbi) [inline openedx/features/edly/utils.py:310]:
```python
edly_user_info_cookie = request.COOKIES.get(settings.EDLY_USER_INFO_COOKIE_NAME, None)
edx_org = get_edx_org_from_cookie(edly_user_info_cookie)

filtered_courses = [course for course in all_courses if course.org == edx_org]
```

**Review by zubair-arbi — CHANGES_REQUESTED** [review]
I have added my feedback.

File: openedx/features/edly/utils.py [hunk 4]
```diff
@@ -290,3 +290,23 @@ def user_can_login_on_requested_edly_organization(request, user):
         return True
 
     return False
+
+
+def filter_courses_based_on_org(request, all_courses):
+    """
+    Filter courses based on the requested URL site.
```
Comment (zubair-arbi) [inline openedx/features/edly/utils.py:297]: nit: new line after first summary line, above `Arguments` and `Returns`.

File: openedx/features/edly/tests/test_utils.py [hunk 5]
```diff
@@ -297,3 +303,29 @@ def test_edly_panel_user_has_edly_org_access(self):
...
+        )
+        response = cookies_api.set_logged_in_edly_cookies(
+            self.request, HttpResponse(),
+            self.user,
+            cookie_settings(self.request)
+        )
+        self._copy_cookies_to_request(response, self.request)
+        courses = CourseFactory.create_batch(5, org=edx_org.short_name)
+
+        filtered_courses = filter_courses_based_on_org(self.request, courses)
```
Comment (zubair-arbi) [inline openedx/features/edly/tests/test_utils.py:325]: We need to add test:

1. create course with `org_1`
2. create  a new course with `org_2`
3. Verify that total courses count is 2
4. Link and login the user with `org_1`
5. Verify that `filter_courses_based_on_org` method returns only one course
6. Verify that `returned course org == user's linked edx org`

**Review by zubair-arbi — CHANGES_REQUESTED** [review]
Added some more feedback.

File: cms/djangoapps/contentstore/views/course.py [hunk 6]
```diff
@@ -514,6 +514,7 @@ def filter_ccx(course_access):
     staff_courses = UserBasedRole(request.user, CourseStaffRole.ROLE).courses_with_role()
     site_courses = UserBasedRole(request.user, GlobalCourseCreatorRole.ROLE).courses_with_role()
     all_courses = filter(filter_ccx, instructor_courses | staff_courses | site_courses)
+    all_courses = filter_courses_based_on_org(request, list(all_courses))
```
Comment (zubair-arbi) [inline cms/djangoapps/contentstore/views/course.py:517]: we don't need to convert `all_courses` to list here
`all_courses = filter_courses_based_on_org(request, all_courses)`

**Review by asadmanzoor93 — APPROVED** [review]

File: openedx/features/edly/tests/test_utils.py [hunk 7]
```diff
@@ -297,3 +304,34 @@ def test_edly_panel_user_has_edly_org_access(self):
         edly_panel_user_group = GroupFactory(name=settings.EDLY_PANEL_USERS_GROUP)
         self.request.user.groups.add(edly_panel_user_group)
         assert edly_panel_user_has_edly_org_access(self.request)
+
+    def test_filter_courses_based_on_org(self):
+        """
+        Test course are filtered properly on current site organization.
+        """
+        edx_org1 = OrganizationFactory()
```
Comment (zubair-arbi) [inline openedx/features/edly/tests/test_utils.py:312]: `edx_org_1`, `edx_org_2`

File: openedx/features/edly/tests/test_utils.py [hunk 8]
```diff
@@ -297,3 +304,34 @@ def test_edly_panel_user_has_edly_org_access(self):
...
+        edx_org2 = OrganizationFactory()
+        courses_of_org1 = CourseFactory.create_batch(2, org=edx_org1.short_name)
```
Comment (zubair-arbi) [inline openedx/features/edly/tests/test_utils.py:314]: `courses_of_org_1`

File: openedx/features/edly/utils.py [hunk 9]
```diff
@@ -290,3 +290,24 @@ def user_can_login_on_requested_edly_organization(request, user):
...
+    Most of our LMS based roles are not organization based we would
+    need to filter courses manually based on org of current site.
+
+    Arguments:
+        request: HTTP request object,
+        all_courses (list): List of CourseAccessRole objects.
+    Returns:
```
Comment (zubair-arbi) [inline openedx/features/edly/utils.py:304]: nit: new line before `Returns:`

File: openedx/features/edly/utils.py [hunk 10]
```diff
@@ -290,3 +290,24 @@ def user_can_login_on_requested_edly_organization(request, user):
...
+
+def filter_courses_based_on_org(request, all_courses):
+    """
+    Filter courses based on the requested URL site.
+    Most of our LMS based roles are not organization based we would
+    need to filter courses manually based on org of current site.
+
+    Arguments:
+        request: HTTP request object,
+        all_courses (list): List of CourseAccessRole objects.
```
Comment (zubair-arbi) [inline openedx/features/edly/utils.py:303]: need to update the definition of `all_courses` kwarg

**Review by zubair-arbi — CHANGES_REQUESTED** [review]
Added more feedback.

**Review by rehan99000 — APPROVED** [review]

**Review by burhandodhy — APPROVED** [review]

File: openedx/features/edly/tests/test_utils.py [hunk 11]
```diff
@@ -33,21 +33,28 @@
...
+    filter_courses_based_on_org,
 )
 from student import auth
 from student.roles import (
     CourseCreatorRole,
     GlobalCourseCreatorRole,
 )
+from xmodule.modulestore.django import modulestore
 from student.tests.factories import UserFactory, GroupFactory
+from xmodule.modulestore.tests.factories import CourseFactory
```
Comment (zubair-arbi) [inline openedx/features/edly/tests/test_utils.py:45]: nit: alphabatically order the imports

File: openedx/features/edly/utils.py (hunk 4 above)
Comment (zubair-arbi) [reply openedx/features/edly/utils.py:297]: still need to add newline after the first summary line

**Review by zubair-arbi — CHANGES_REQUESTED** [review]
Added 2 more small suggestions.

File: openedx/features/edly/tests/test_utils.py (hunk 11 above)
Comment (zubair-arbi) [reply openedx/features/edly/tests/test_utils.py:45]: Still need to address this.

Comment (zubair-arbi) [reply openedx/features/edly/tests/test_utils.py:45]: https://edx.readthedocs.io/projects/edx-developer-guide/en/latest/style_guides/python-guidelines.html#imports-order

**Review by zubair-arbi — APPROVED** [review]
Changes look good to me now. Please rebase/squash your PR before merging.

**Pull request merged**